core/tests/data/ser/json.cpp

```diff
@@ -0,0 +1,93 @@
+#include <doctest/doctest.h>
+#include <glm/vec3.hpp>
+
+#include <cubos/core/data/ser/debug.hpp>
+#include <cubos/core/data/ser/json.hpp>
+#include <cubos/core/reflection/external/glm.hpp>
+#include <cubos/core/reflection/external/map.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+#include <cubos/core/reflection/external/uuid.hpp>
+#include <cubos/core/reflection/external/vector.hpp>
+#include <cubos/core/reflection/traits/constructible_utils.hpp>
+#include <cubos/core/reflection/traits/fields.hpp>
+
+using cubos::core::data::JSONSerializer;
+using cubos::core::reflection::FieldsTrait;
+using cubos::core::reflection::Type;
+
+#define AUTO_TEST(type, value, expected)                                                                               \
+    ser.write<type>(value);                                                                                            \
+    json = ser.output();                                                                                               \
+    CHECK((json.dump()) == (expected));
+
+TEST_CASE("data::JSONSerializer")
+{
+    JSONSerializer ser{};
+    auto json = ser.output();
+
+    /// [Primitive types]
```

Comment:
```suggestion
    // Primitive types
```
Do the same for the comments below